packages/mongo-nav/src/mongo-select/OrgSelect.tsx

```diff
@@ -192,14 +193,22 @@ function OrgSelect({
 }: OrganizationMongoSelectProps) {
   const [value, setValue] = useState('');
   const [filteredData, setFilteredData] = useState(data);
+  const [isFetching, setIsFetching] = useState(false);
   const [open, setOpen] = useState(false);
   const onElementClick = useOnElementClick();
   const { usingKeyboard: showFocus } = useUsingKeyboardContext();
   const { width: viewportWidth } = useViewportSize();
   const isTablet = viewportWidth < breakpoints.medium;
 
-  const checkPlanType = data?.[0]?.planType;
-  const showPlanType = !data?.every(datum => datum.planType === checkPlanType);
+  const isFiltered = value !== '';
+  const isAdminSearch = isFiltered && admin && mode === Mode.Production;
+
+  const renderedData = isFiltered ? filteredData : data;
+
+  const checkPlanType = renderedData?.[0]?.planType;
+  const showPlanType = !renderedData?.every(
+    datum => datum.planType === checkPlanType,
+  );
```

Comment:
I'm hoping this reduces the code paths, with `renderedData` ultimately being what's rendered.